fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneAutoCompleteResultCursor.java

```diff
@@ -177,7 +177,7 @@ private void performLookup() throws IOException {
         }
         long startTime = System.nanoTime();
 
-        lookupResults = lookup().skip(skip);
+        lookupResults = lookup().skip(skip).limitRowsTo(limit);
```

Comment:
I'm a little surprised that this passes the tests. This looks right to me, but the fact that none of the Lucene tests failed seems to suggest that we don't have any tests that validate that the limit is enforced, which could be a sign of a gap in our coverage.

Reply:
I also suspect that
```
searcher.search(finalQuery, limit)
```
in `lookup` is failing to take account of `skip` properly.

Reply:
@MMcM @alecgrieser  fixed the skip for ```searcher.search```. And added tests to cover this. The new added tests failed without this fix.